### ISSUE TITLE
Extrapolate with missing does not work

### ISSUE THREAD

**Issue by ohmsweetohm1** [JuliaMath/Interpolations.jl#370]
```julia
import Interpolations

x = collect(1.0:1.0:40.0)
y = convert(Vector{Union{Float64,Missing}}, rand(length(x)))
y[5:10] .= missing

idx_valid = findall(yi -> !ismissing(yi), y)
xf = convert.(Float64, x[idx_valid])
yf = convert.(Float64, y[idx_valid])

itp = Interpolations.interpolate((xf,), yf, Interpolations.Gridded(Interpolations.Linear())) 
itp = Interpolations.extrapolate(itp, missing)
```
```julia
xi = collect(10.0:1.0:40.0)
yi = itp.(xi)

31-element Array{Float64,1}:
 0.8696268062771625
 0.9426581118022204
 0.7500886600236532
 0.6381631167258599
 0.6679414190779305
 0.41037321112298075
 ⋮
 0.8469243934808046
 0.294974616523348
 0.18761139198378007
 0.19510822371782832
 0.4671396244747503
```

```julia
xi = collect(10.0:1.0:50.0)
yi = itp.(xi)

ERROR: MethodError: Cannot `convert` an object of type Missing to an object of type Float64
```

**Comment by MatFi** [comment]
I don't see the use of such functionality, already due to type stability alone you should avoid such things and try if `NaN` can fulfill the same purpose. 

```
julia> NaN isa Float64
true

julia> missing isa Float64
false
```
What exactly is the reason you want `missing` to work here?

**Comment by ohmsweetohm1** [comment]
`NaN` is not the same as `missing`.

From Julia:
> Julia provides support for representing missing values in the statistical sense, that is for situations where no value is available for a variable in an observation, but a valid value theoretically exists.

If I have a timeseries and want to extrapolate a value outside the known domain it should be possible to yield missing.

**Comment by ohmsweetohm1** [comment]
Also: under http://juliamath.github.io/Interpolations.jl/latest/control/#Gridded-interpolation-1 it is stated:
> missing data will naturally propagate through the interpolation, where some values will become missing.

Why is that different for extrapolation?

**Comment by MatFi** [comment]
Interesting, indeed  looks like a bug also for me now.
As a workaround you can overload the corresponding function. Is this worth a pull request @mkitti ?

```julia
import Interpolations
@inline function (etp::Interpolations.FilledExtrapolation{T,N})(x::Vararg{Number,N}) where {T,N}
    itp = Interpolations.parent(etp)
    coefs = Interpolations.coefficients(itp)
    Tret = typeof(prod(x) * zero(eltype(coefs)))
    if checkbounds(Bool, itp, x...)
        @inbounds itp(x...)
    else
        #convert(Tret, etp.fillvalue) #this causes the problem
        #promote_type(Tret,typeof(etp.fillvalue)) #is probably better but already done by the constructor
        etp.fillvalue # so this should be enougth
    end
end
x = collect(1.0:1.0:40.0)
y = convert(Vector{Union{Float64,Missing}}, rand(length(x)))
y[5:10] .= missing

idx_valid = findall(yi -> !ismissing(yi), y)
xf = convert.(Float64, x[idx_valid])
yf = convert.(Float64, y[idx_valid])

itp = Interpolations.interpolate((xf,), yf, Interpolations.Gridded(Interpolations.Linear())) 
itp = Interpolations.extrapolate(itp, missing)

julia> yi = itp.(xi)
51-element Array{Union{Missing, Float64},1}:
 0.22054370978813284
 0.24669603447950794
 0.0310143527164084
 0.6891753629560986
 0.4028200345855397
 ⋮
  missing
  missing
  missing
  missing

julia> 
```

**Comment by mkitti** [comment]
@MatFi pull requests are always welcome

**Comment by MatFi** [comment]
Now I see that this could be addressed/fixed in #367
